disk/tests/test_e2e.py

```diff
@@ -11,4 +11,5 @@
 def test_check(dd_agent_check):
     aggregator = dd_agent_check()
     for metric in common.EXPECTED_METRICS:
-        aggregator.assert_metric(metric['metric'], metric_type=aggregator.GAUGE, device=metric['device'])
+        for device in common.EXPECTED_DEVICE:
+            aggregator.assert_metric(metric, metric_type=aggregator.GAUGE, device=device)
```

Comment:
(nit) Could you also add:

```
aggregator.assert_all_metrics_covered()
aggregator.assert_metrics_using_metadata(get_metadata_metrics())
```

Reply:
I noticed recently that on the Azure CI environment, there is a device `/dev/sdb1` that appears but I'm unable to reproduce locally. Adding `aggregator.assert_all_metrics_covered()` might fail depending on platform.

Reply:
You can add an extra assert with `aggregator.assert_metric("the metric", at_least=0)`. That's the workaround we use in other similar situation.